app/controllers/manager/statistics/employees_controller.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Manager
+  module Statistics
+    class EmployeesController < ApplicationController
+      layout false
+
+      def index; end
+
+      def academic
+        @series = Employee.active.academic.group('titles.name').count
```

Comment:
Yada:

```ruby
Rails.cache.fetch(SOME_CACHE_KEY, expires: 1.day)
  @series = Employee.active.academic.group('titles.name').count
end
```

Reply:
Böyle bir cache mekanizması kurulabilir. Zaten artık kullanmayada başlamak lazım.

Reply:
Cok real time bir veriye ihtiyac yoksa bu count'lari database'de de tutabiliriz. Daha once User modelinde `articles_count` ve `projects_count` icin boyle yapmistik. Her gece calisan bir cron gorevi bu count sayilarini guncelleyecekti plana gore, ama dokku'da cron sorun cikartinca sanirim o tarafi yapamamistik.

Reply:
Dokku cron konusunu tekrar gündeme alacağız.  @ecylmz Yanılmıyorsam orada son karar [sidekiq-cron](https://github.com/ondrejbartas/sidekiq-cron) idi.  PR kapatıldıktan sonra ayrıca tartışırız.

Reply:
Bana bu tür durumlarda cache kullanmak daha mantıklı geliyor. Cron veya bir task oluşturup onu çalıştırmaktansa, doğru bir şekilde `cache` yönetmek daha kolay, ama her iki yönteminde kullanılacağı yerler var (Bknz: User modeli). Cron görevlerinin bizim için bir dezavantajı ise ,sende bahsetmişsin dokku tarafında sorun çıkartıyor olması. Bir de `*_count` `User` modeli için konuşmak gerekirse bir kullanıcının projesi sayısı ve makale sayısını tutmak için kullanılıyor. Burada ise sistem genelinde akademik personel sayısı isteniyor.  Bunun içinde ekstra bir model oluşturmak ve verileri satır satır veya tek satır çok sütunlu olarak tutmak gerekecek. Son olarak bu sayfa şimdilik çok aşırı kullanımda olmayacak o yüzden performans sorunları birazcık ötelenebilir.

Reply:
Cache vs cron konusunda dediğine katılıyorum.  Cache ile çözülmesi daha şık olur.  Fakat:

> Cron görevlerinin bizim için bir dezavantajı ise ,sende bahsetmişsin dokku tarafında sorun çıkartıyor olması.

Teknik olarak bu bir kaçınma nedeni olmamalı.  Zamanlanmış görevlere her türlü ihtiyacımız var.  Bazı problemler en basit veya anlamlı olarak cron ile çözülür, ör. periyodik temizlikler, uygulamanın yeni bir döneme hazırlanması gibi.  Bunun öyle bir problem olduğunu söylemiyorum.  "Cron konusunu bir an önce çözelim" sinyali vermek için yazdım.